db/memory/src/main/java/org/commonjava/indy/mem/data/MemoryStoreDataManager.java

```diff
@@ -354,11 +357,69 @@ public void clear( final ChangeSummary summary )
     @Override
     public RemoteRepository findRemoteRepository( final String url )
     {
+
         List<Map.Entry<StoreKey, ArtifactStore>> copy = new ArrayList<>( stores.entrySet() );
 
+        /*
+           This filter do these things:
+             * First compare ip, if ip same, and the path(without last slash) same too, the repo is found
+             * If ip not same, then compare the url without scheme and last slash (if has) to find the repo
+         */
+        Predicate<Map.Entry<StoreKey, ArtifactStore>> findingRepoFilter = e -> {
+            if ( ( remote == e.getValue().getKey().getType() ) )
+            {
+                final UrlInfo urlInfo = new UrlInfo( url );
+                final String targetUrl = ( (RemoteRepository) e.getValue() ).getUrl();
+                final UrlInfo targetUrlInfo = new UrlInfo( targetUrl );
+                final String ipForUrl;
+                final String ipForTargetUrl;
+                try
+                {
+                    ipForUrl = urlInfo.getIpForUrl();
+                    ipForTargetUrl = targetUrlInfo.getIpForUrl();
+                }
+                catch ( UnknownHostException ue )
+                {
+                    logger.warn( "Failed to filter remote: ip fetch error.", ue );
+                    return false;
+                }
+
+                if ( ipForUrl != null && ipForUrl.equals( ipForTargetUrl ) )
+                {
+                    if ( urlInfo.getFileWithNoLastSlash().equals( targetUrlInfo.getFileWithNoLastSlash() ) )
+                    {
+                        logger.debug( "Repository found because of same ip, url is {}, store key is {}", url,
+                                      e.getValue().getKey() );
+                        return true;
+                    }
+                }
+                else
+                {
+                    logger.debug( "ip not same: ip for url:{}-{}; ip for searching repo: {}-{}", url, ipForUrl,
+                                  e.getValue().getKey(), ipForTargetUrl );
+                    try
+                    {
+                        if ( urlInfo.getUrlWithNoSchemeAndLastSlash()
+                                    .equals( targetUrlInfo.getUrlWithNoSchemeAndLastSlash() ) )
+                        {
+                            logger.debug( "Repository found because of same host, url is {}, store key is {}", url,
+                                          e.getValue().getKey() );
+                            return true;
+                        }
+                    }
+                    catch ( IllegalArgumentException error )
```

Comment:
Oops, this is a refactoring missing. This IllegalArgumentException is used to avoid UrlInfo construction failure caused by MalformedURLException (see https://github.com/Commonjava/indy/blob/master/api/src/main/java/org/commonjava/indy/util/UrlInfo.java#L47), but pointed to wrong place here when I did some refactoring. I think this IllegalArgumentException should not block the comparison of the filtering, so just catch here and give an error log.